tests/xharness/Jenkins/TestServer.cs

```diff
@@ -16,14 +16,15 @@ class TestServer {
 
 		public Task RunAsync (Jenkins jenkins, HtmlReportWriter htmlReportWriter)
 		{
-			var server = new HttpListener ();
+			HttpListener server;
 
 			// Try and find an unused port
 			int attemptsLeft = 50;
 			int port = 51234; // Try this port first, to try to not vary between runs just because.
 			Random r = new Random ((int) DateTime.Now.Ticks);
-			while (attemptsLeft-- > 0) {
+			do {
 				var newPort = port != 0 ? port : r.Next (49152, 65535); // The suggested range for dynamic ports is 49152-65535 (IANA)
+				server = new HttpListener ();
 				server.Prefixes.Clear ();
 				server.Prefixes.Add ("http://*:" + newPort + "/");
 				try {
```

Comment:
I think you're going to need a `Dispose` call on HttpListener in the catch block so as to not leak resources.

Reply:
@stephen-hawley I thought the same thing, but it turns out there is no Dispose method on HttpListener: https://sharplab.io/#v2:EYLgtghglgdgNAFxFANgHwAICYCMBYAKAwGYACbUgYVIG9DSHyyMAWUgWQAoBKW+xgQDcIAJ1IooAZwQBTGDLEBeUvIDu5HADoAcjISaAEggQAHADJTZ8sTwDc/AQwnS5CzQBEpJgPaSZpOwdSAF9CYKA===

In fact HttpListener implements IDisposable, but it implements IDisposable.Dispose  internally, and the [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.net.httplistener.system-idisposable-dispose?view=net-8.0) says "Applications should use the [Close](https://learn.microsoft.com/en-us/dotnet/api/system.net.httplistener.close?view=net-8.0) method instead of calling this method." - but that's not necessary, because the listener never started.

Also note that the HttpListener is _already disposed_ - because you get an ObjectDisposedException if you try to use it (see the PR's description).